BaseTools/Source/Python/Workspace/DscBuildData.py

```diff
@@ -2957,13 +2963,17 @@ def GenerateByteArrayValue (self, StructuredPcds):
 
         # start generating makefile
         MakeApp = PcdMakefileHeader
-        if sys.platform == "win32":
-            MakeApp = MakeApp + 'APPFILE = %s\\%s.exe\n' % (self.OutputPath, PcdValueInitName) + 'APPNAME = %s\n' % (PcdValueInitName) + 'OBJECTS = %s\\%s.obj %s.obj\n' % (self.OutputPath, PcdValueInitName, os.path.join(self.OutputPath, PcdValueCommonName)) + 'INC = '
+        if sys.platform== "win32":
+            if(os.environ.get('CLANG_MAKEFILE_TYPE')!='gnu'):
+                MakeApp = MakeApp + 'APPFILE = %s\\%s.exe\n' % (self.OutputPath, PcdValueInitName) + 'APPNAME = %s\n' % (PcdValueInitName) + 'OBJECTS = %s\\%s.obj %s.obj\n' % (self.OutputPath, PcdValueInitName, os.path.join(self.OutputPath, PcdValueCommonName)) + 'INC = '
+            else:
+                MakeApp = MakeApp + PcdGccMakefile
+                MakeApp = MakeApp + 'APPFILE = %s\\%s.exe\n' % (self.OutputPath, PcdValueInitName) + 'APPNAME = %s\n' % (PcdValueInitName) + 'OBJECTS = %s\\%s.o %s.o\n' % (self.OutputPath, PcdValueInitName, os.path.join(self.OutputPath, PcdValueCommonName)) + \
+                'include $(MAKEROOT)\\Makefiles\\app.makefile\n' + 'TOOL_INCLUDE +='

```

Comment:
use os.pathsep and suffix variables for {}/.exe and .o/.obj so the MakeApp assignment is common.